e2e/pull/pull.go

```diff
@@ -464,13 +465,107 @@ func (c ctx) testPullDisableCacheCmd(t *testing.T) {
 	}
 }
 
+// testPullUmask will run some pull tests with different umasks, and
+// ensure the output file hase the correct permissions.
+func (c ctx) testPullUmask(t *testing.T) {
+	umaskTests := []struct {
+		name       string
+		imagePath  string
+		umask      int
+		expectPerm uint32
+		force      bool
+	}{
+		{
+			name:       "0022 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0022-umask-pull"),
+			umask:      0022,
+			expectPerm: 0755,
+		},
+		{
+			name:       "0077 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0077-umask-pull"),
+			umask:      0077,
+			expectPerm: 0700,
+		},
+		{
+			name:       "0027 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0027-umask-pull"),
+			umask:      0027,
+			expectPerm: 0750,
+		},
+
+		// With the force flag.
+		{
+			name:       "0022 umask pull",
+			imagePath:  filepath.Join(c.env.TestDir, "0022-umask-pull"),
+			umask:      0022,
+			expectPerm: 0755,
```

Comment:
These tests with force (overwriting) aren't really adding anything I don't think, since they are overwriting files in a way that the exact same permissions are expected / pretty much guaranteed.

They *would* be testing something valuable if the file they were overwriting *didn't already* have the expected permissions.

Reply:
That is... assuming we *want* the file permissions to be changed on overwrite.

Reply:
Good point, I'll come up with a fix...